src/main/java/org/jenkinsci/plugins/prometheus/service/DefaultPrometheusMetrics.java

```diff
@@ -1,54 +1,74 @@
 package org.jenkinsci.plugins.prometheus.service;
 
 import hudson.ExtensionList;
+import hudson.init.InitMilestone;
+import hudson.init.Initializer;
 import io.prometheus.client.Collector;
 import io.prometheus.client.CollectorRegistry;
 import io.prometheus.client.dropwizard.DropwizardExports;
 import io.prometheus.client.exporter.common.TextFormat;
 import io.prometheus.client.hotspot.DefaultExports;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.concurrent.atomic.AtomicReference;
 import jenkins.metrics.api.Metrics;
-import org.jenkinsci.plugins.prometheus.*;
+import org.jenkinsci.plugins.prometheus.CodeCoverageCollector;
+import org.jenkinsci.plugins.prometheus.DiskUsageCollector;
+import org.jenkinsci.plugins.prometheus.ExecutorCollector;
+import org.jenkinsci.plugins.prometheus.JenkinsStatusCollector;
+import org.jenkinsci.plugins.prometheus.JobCollector;
 import org.jenkinsci.plugins.prometheus.config.disabledmetrics.FilteredMetricEnumeration;
 import org.jenkinsci.plugins.prometheus.util.JenkinsNodeBuildsSampleBuilder;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.io.IOException;
-import java.io.StringWriter;
-import java.util.concurrent.atomic.AtomicReference;
-
+@Restricted(NoExternalUse.class)
 public class DefaultPrometheusMetrics implements PrometheusMetrics {
 
     private static final Logger logger = LoggerFactory.getLogger(DefaultPrometheusMetrics.class);
 
+    private volatile boolean initialized = false, initializing = false;
```

Comment:
Improve readability per suggestion: 

```suggestion
    private volatile boolean initialized = false;
    private volatile boolean initializing = false;
```